oak_sev_snp_attestation_report/src/lib.rs

```diff
@@ -0,0 +1,349 @@
+// Copyright 2024 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+//! AMD SEV-SNP data structures for attestation reports.
+
+// TODO(#3703): Remove when fixed.
+#![allow(clippy::extra_unused_type_parameters)]
+#![no_std]
+
+use bitflags::bitflags;
+use strum::FromRepr;
+use zerocopy::{AsBytes, FromBytes, FromZeroes};
+
+/// A signed attestation report.
+///
+/// See Table 22 in <https://www.amd.com/system/files/TechDocs/56860.pdf>.
+#[repr(C)]
+#[derive(Debug, AsBytes, FromZeroes, FromBytes)]
+pub struct AttestationReport {
+    /// The data contained in the report.
+    pub data: AttestationReportData,
+    /// The signature over the data.
+    pub signature: EcdsaSignature,
+}
+static_assertions::assert_eq_size!(AttestationReport, [u8; 1184]);
+
+impl AttestationReport {
+    /// Creates a new AttestationReport with all zeros and the provided value on the
+    /// data.custom_report_data field.
+    pub fn from_custom_report_data(custom_report_data: [u8; REPORT_DATA_SIZE]) -> Self {
```

Comment:
nit: I would remove "custom" from the names since report_data is always custom

Reply:
Done, apologies, that was a leftover from abandoned refactor to rename.